### PR TITLE
HTS-1039: Make sure we're using MDCExecutionContext not global Execut…

### PR DESCRIPTION
**Pull request by sureshhmrc**
…ionContext

### CODE REVIEW

File: app/uk/gov/hmrc/helptosave/services/UserCapService.scala [hunk 1]
```diff
@@ -24,23 +24,22 @@ import uk.gov.hmrc.helptosave.config.AppConfig
...
+  def isAccountCreateAllowed()(implicit ec: ExecutionContext): Future[UserCapResponse]
 
-  def update(): Future[Unit]
+  def update()(implicit ec: ExecutionContext): Future[Unit]
 
 }
 
 @Singleton
-class UserCapServiceImpl @Inject() (userCapStore: UserCapStore)(implicit appConfig: AppConfig) extends UserCapService with Logging {
+class UserCapServiceImpl @Inject() (userCapStore: UserCapStore)(implicit appConfig: AppConfig) extends UserCapService with Logging with WithMdcExecutionContext {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/services/UserCapService.scala]: do we need to extend with `WithMdcExecutionContext` here?

File: app/uk/gov/hmrc/helptosave/controllers/StrideController.scala [hunk 2]
```diff
@@ -27,7 +27,7 @@ import uk.gov.hmrc.helptosave.config.AppConfig
 import uk.gov.hmrc.helptosave.connectors.PayePersonalDetailsConnector
 import uk.gov.hmrc.helptosave.repo.EnrolmentStore
 import uk.gov.hmrc.helptosave.services.EligibilityCheckService
-import uk.gov.hmrc.helptosave.util.LogMessageTransformer
+import uk.gov.hmrc.helptosave.util.{LogMessageTransformer, WithMdcExecutionContext}
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/controllers/StrideController.scala]: is this import being used?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/controllers/StrideController.scala]: yes StrideController extends WithMdcExecutionContext

File: app/uk/gov/hmrc/helptosave/services/UserCapService.scala (hunk 1 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/services/UserCapService.scala]: I think I again missed pushing the commit - let me check

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/services/UserCapService.scala]: pushed the commit

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**